### PR TITLE
[SWUI-46][FEAT] - Improve Dialog component accessibility

### PR DESCRIPTION
**Pull request by ElRodrigote**
## Fixes: [SWUI-46](https://linear.app/swaprhq/issue/SWUI-46/improve-dialog-component-accessibility)

# Description
* Install `VisuallyHidden` from Radix to hide DialogTitle and DialogDescription from the UI if needed
* Export Radix `Description` as `DialogDescription`
* Updates `@headlessui/react` dependency in our `ui` package due to a build error (see [this issue](https://github.com/tailwindlabs/headlessui/issues/3167), solved in the [PR #3254](https://github.com/tailwindlabs/headlessui/releases/tag/%40headlessui%2Freact%40v2.1.0))

### CODE REVIEW

File: apps/website/app/ui/page.tsx [hunk 1]
```diff
@@ -649,6 +651,11 @@ export default function UI() {
                 <DialogHeader>
                   <DialogTitle>Hi I'm a Modal title</DialogTitle>
                 </DialogHeader>
+                <VisuallyHidden asChild>
+                  <DialogDescription className="px-4">
+                    Here a really long text will be displayed.
+                  </DialogDescription>
+                </VisuallyHidden>
```
Comment (berteotti) [inline apps/website/app/ui/page.tsx:658]: Shouldn't we add `VisuallyHidden` to `DialogDescription` and `DialogTitle`??

Comment (berteotti) [reply apps/website/app/ui/page.tsx:658]: I mean, add it to the component it self. Cause we are using it everywhere and looks like every `DialogTitle` need a `VisuallyHidden` component. So probably we should wrap the component with `VisuallyHidden` on Dialog file. What do you think?

Comment (ElRodrigote) [reply apps/website/app/ui/page.tsx:658]: Mmm, I'm not sure if I'm understanding correctly. 

Do you mean to modify how we export `DialogTitle` to be wrapped with `VisuallyHidden` by default? 
If that's the case then we would be hiding all our already visible DialogTitles in Presagio for example:
[Preagio's ConfirmTrade component](https://github.com/SwaprHQ/presagio/blob/7536ca6e60416fa6c8bc226d8597c0609717ef2b/app/components/ConfirmTrade.tsx#L230)
[Presagio's EmbedMarket component](https://github.com/SwaprHQ/presagio/blob/7536ca6e60416fa6c8bc226d8597c0609717ef2b/app/components/EmbedMarketModal.tsx#L99)

We don't need to render the title in the TransactionModal or similar components.
I would:
* Export it as-is from our lib
* Let the clients from this package install the `VisuallyHidden` component and decide whether use it, or implement another solution to hide the component
* Maybe do some conditional rendering like:
```tsx
hidden ? (
  <VisuallyHidden asChild>
    <DialogDescription {...props}>
      {description}
    </DialogDescription>
  </VisuallyHidden>
) : (
  <DialogDescription {...props}>
   {description}
  </DialogDescription>
)
```

Comment (berteotti) [reply apps/website/app/ui/page.tsx:658]: I now understand how it works, let's not to that! 😄 

But we can export `VisuallyHidden` from Dialog file maybe? And we don't need to had a new lib to presagio package.json.

**Review by berteotti — APPROVED** [review]

**Pull request merged**